src/celengine/starcolors.h

```diff
@@ -32,15 +40,26 @@ class ColorTemperatureTable
             return colors[colorTableIndex];
     }
 
+    Color lookupTintColor(float temp, float saturation) const
+    {
+        Eigen::Vector3f color = lookupColor(temp).toVector3();
+        color = (1.0f - saturation) * Eigen::Vector3f::Ones() + saturation * color;
```

Comment:
```
color = vecgl::mix(Eigen::Vector3f::Ones(), color, saturation);
```

Reply:
annoyingly that doesn't work at the moment because `Eigen::Vector3f::Ones()` actually returns `Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, Eigen::Vector3f>` rather than `Eigen::Vector3f`, so the template matching fails

Reply:
fixed... just needed a bit more template magic